template.yaml

```diff
@@ -110,13 +110,33 @@ Resources:
           Keep Model running for 1 hr:
             Type: Wait
             Seconds: 3540
-            Next: Disable SQS Trigger
+            Next: Check Queue Again
+          Check Queue Again:
+            Type: Task
+            Resource: !GetAtt SQSPollerFunction.Arn
+            ResultPath: $.moremessagesinqueue
+            Next: Are there more images?
+          Are there more images?:
+            Type: Choice
+            Choices:
+              - Variable: $.moremessagesinqueue
+                StringEquals: stop
+                Next: Disable SQS Trigger
+            Default: Keep Model running for 1 hr
           Enable SQS Trigger:
             Type: Task
+            ResultPath: $.alreadyrunning
             Parameters:
               - Action: enable
             Resource: !GetAtt ToggleTriggerFunction.Arn
-            Next: Keep Model running for 1 hr
+            Next: Is another machine already running?
```

Comment:
I haven't tested this, I guess you already did some manual testing to check if the circuit breaker works?